### ISSUE TITLE
Solution for 'intentional' errors

### ISSUE THREAD

**Issue by natashawatkins** [QuantEcon/lecture-source-py#85]
We need a way to label cells that produce errors, without having the errors count towards coverage (https://lectures.quantecon.org/status.html). Possible we could introduce a `skip-test` label.

**Comment by natashawatkins** [comment]
@mmcky thinking about this, I think we need both a `skip-test` label for errors we want to display, and a `no-execute` label for code we don't want run (ie. C code), which should also no count towards the testing stats

**Comment by mmcky** [comment]
@natashawatkins thinking this through over the past through days. I think it might be best to use.

- [ ] ``:allow-error:`` for cell-blocks that should execute (when compiling the website) but don't execute when running the `coverage` test suite. In the future we could add error types such as ``ValueError`` to catch only specific error types etc. For now it will be just a boolean flag. It is also pretty self explanatory.

If there are blocks that are not the same as the `default` language (i.e. python3) then they should be added as markdown blocks anyway in ``jupinx`` conversion.

**Comment by mmcky** [comment]
This PR https://github.com/QuantEcon/sphinxcontrib-jupyter/pull/129 implements the SHORT TERM solution for this issue. The Makefile for ``quantecon.build.lectures` has also been updated.

**Comment by natashawatkins** [comment]
So at the moment we won't be displaying errors? Does that mean that we need to manually put them in?

**Comment by mmcky** [comment]
@natashawatkins not sure what you mean here. This allows for those `:no-execute:` statements to be ignored during the compilation of the notebook so they will be in execute blocks rather than markdown blocks.

**Comment by natashawatkins** [comment]
I'm confused - there are two separate cases:
1. Code that intentionally has an error and we want to display the error output.
2. Code that should not be executed - we don't want to display the error. Eg. a partial solution to an exercise, see here: https://lectures.quantecon.org/py/python_advanced_features.html#exercise-2

`no-execute` should be used in the second case - is that right?

**Comment by mmcky** [comment]
Ah so there are really 3 cases. 

I was working to:

1. Same as your `1` above
2. Show code such as `C` code in a python lecture. 

This `2` case is handed by `Jupinx`:

   Also if there are blocks that are not the same as the default language (i.e. python3) then they should be added as markdown blocks anyway in jupinx conversion.

but there is a third case such as your `2` above. Bummer :) Can you think of any others?

**Comment by natashawatkins** [comment]
So what should we do in the first case? Will jupinx insert the error output and not count the error towards the testing stats?

And not that I can think of.

**Comment by mmcky** [comment]
- [x] added ability to globally `skip-test` and updated `quantecon.build.lectures` and `jupinx` (https://github.com/QuantEcon/sphinxcontrib-jupyter/pull/131)

**Comment by natashawatkins** [comment]
Is @fkazemian available to work on this maybe? I can check the PRs

**Comment by fkazemian** [comment]
Hi @natashawatkins  I will do it

**Comment by fkazemian** [comment]
Hi @mmcky 
I think in here we have to use
`skip-test for anything that should be executed (i.e. purposeful error) but shouldn't be counted in any coverage tests`
![screenshot from 2018-10-15 13-28-47](https://user-images.githubusercontent.com/20714542/46927735-b1ac8700-d082-11e8-9914-3d0bba3c8c47.png)
because we want it to be executed, but when I'm compiling this code:

```

.. code-block:: python3
    :class: skip-test

    f = open('us_cities.txt')
    f.__next__()
    
.. code-block:: none
    :class: skip-test
    
    'new york: 8244910\n'
    
.. code-block:: python3
    :class: skip-test

    f.__next__()

```
but it's not working, Do I use a right tag?
![screenshot from 2018-10-15 14-02-25](https://user-images.githubusercontent.com/20714542/46927789-018b4e00-d083-11e8-9ffb-8ce3ba185892.png)

**Comment by mmcky** [comment]
for `skip-test` to work you will need to latest version of `jupinx` (from `master`). This was only merged yesterday. If you have the latest version please let me know and I will take a closer look.

**Comment by natashawatkins** [comment]
@fkazemian could you also do this issue in separate PRs for each lecture? thanks

**Comment by fkazemian** [comment]
excuse me Have you tested the latest version of jupinx locally?
I don't know why mine is like 
![screenshot from 2018-10-17 10-56-10](https://user-images.githubusercontent.com/20714542/47054015-64075a00-d1fb-11e8-80cd-52880d6327ad.png)

**Comment by mmcky** [comment]
hi @fkazemian if you are not using `quanecon.build.lectures` to compile then it is because the theme has not been applied to the generated output.

**Comment by fkazemian** [comment]
Hi @mmcky Actually I'm using 'quanecon.build.lectures' and I used git pull origin master

**Comment by fkazemian** [comment]
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/python_advanced_features.rst:.. code-block:: none
- [ ] ./rst_files/schelling.rst:.. code-block:: none
- [ ] ./rst_files/schelling.rst:.. code-block:: none
- [ ] ./rst_files/python_essentials.rst:.. code-block:: none
- [ ] ./rst_files/python_essentials.rst:.. code-block:: none
- [ ] ./rst_files/python_essentials.rst:.. code-block:: none
- [ ] ./rst_files/pandas.rst:.. code-block:: none
- [ ] ./rst_files/pandas.rst:.. code-block:: none
- [ ] ./rst_files/debugging.rst:.. code-block:: none
- [ ] ./rst_files/debugging.rst:.. code-block:: none
- [ ] ./rst_files/debugging.rst:.. code-block:: none
- [ ] ./rst_files/finite_markov.rst:.. code-block:: none
- [ ] ./rst_files/python_by_example.rst:.. code-block:: none
- [ ] ./rst_files/oop_intro.rst:.. code-block:: none

**Comment by fkazemian** [comment]
- [x]  ./rst_files/stationary_densities.rst: :class: no-execute
- [x] ./rst_files/stationary_densities.rst: :class: no-execute 
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_oop.rst: :class: no-execute 
- [ ] ./rst_files/python_oop.rst: :class: no-execute 

- [x] ./rst_files/python_essentials.rst: :class: no-execute  #98
- [x] ./rst_files/python_essentials.rst: :class: no-execute
![screenshot from 2018-10-26 11-16-09](https://user-images.githubusercontent.com/20714542/47537295-ab36ce80-d910-11e8-9ae9-9c19248d324f.png)
```
In this case you can shift the file to the pwd or specify the `full path <https://en.wikipedia.org/wiki/Path_%28computing%29>`_ to the file

.. code-block:: python3
    :class: no-execute

    f = open('insert_full_path_to_file/newfile.txt', 'r')
```

- [x] ./rst_files/python_essentials.rst: :class: no-execute 
![screenshot from 2018-10-26 11-19-52](https://user-images.githubusercontent.com/20714542/47537383-26988000-d911-11e8-92ee-76843fd5d045.png)
```
If you did the exercises in the :ref:`previous lecture <python_by_example>`, you would have come across the statement

.. code-block:: python3
    :class: no-execute

    plt.plot(x, 'b-', label="white noise")

In this call to Matplotlib's ``plot`` function, notice that the last
```
- [x] ./rst_files/debugging.rst: :class: no-execute 
![screenshot from 2018-10-26 10-29-01](https://user-images.githubusercontent.com/20714542/47536084-1a5cf480-d90a-11e8-9bbe-6140d54dd566.png)
```
Let's consider a simple (and rather contrived) example

.. code-block:: python3
    :class: no-execute

    import numpy as np
    import matplotlib.pyplot as plt

    def plot_log():
        fig, ax = plt.subplots(2, 1)
        x = np.linspace(1, 2, 10)
        ax.plot(x, np.log(x))
        plt.show()

    plot_log()  # Call the function, generate plot


This code is intended to plot the ``log`` function over the interval
```
- [x] ./rst_files/debugging.rst: :class: no-execute 
![screenshot from 2018-10-26 10-38-03](https://user-images.githubusercontent.com/20714542/47536382-5c3a6a80-d90b-11e8-8a7f-79c41e1cc1a4.png)
---------------------------------
- [x] ./rst_files/debugging.rst: :class: no-execute 
- [x] ./rst_files/debugging.rst: :class: no-execute 
- [x] ./rst_files/debugging.rst: :class: no-execute
- [x] ./rst_files/debugging.rst: :class: no-execute 
![screenshot from 2018-10-26 10-42-47](https://user-images.githubusercontent.com/20714542/47536494-0d410500-d90c-11e8-9e58-d18b1a791d9d.png)

---------------------------
- [x] ./rst_files/debugging.rst: :class: no-execute 
- [x] ./rst_files/debugging.rst: :class: no-execute
![screenshot from 2018-10-26 10-45-23](https://user-images.githubusercontent.com/20714542/47536542-627d1680-d90c-11e8-9963-a649d44eacbf.png)
-----------------------------
- [x] ./rst_files/sci_libs.rst: :class: no-execute 
![screenshot from 2018-10-26 10-18-31](https://user-images.githubusercontent.com/20714542/47535746-9e15e180-d908-11e8-94ec-4c524eabc396.png)
* we have to use `.. code-block:: C` which is fine here, but it also use `:class: no-execute`, I think we don't want to run it as well, so I leave it unchanged:
```
Here's a C function that will do the same thing

.. code-block:: c
    :class: no-execute

    double geo_prog(double alpha, int n) {
        double current = 1.0;
        double sum = current;
        int i;
        for (i = 1; i <= n; i++) {
            current = current * alpha;
            sum = sum + current;
        }
        return sum;
    }


If you're not familiar with C, the main thing you should take notice of is the
type definitions
```
- [x] ./rst_files/numba.rst: :class: no-execute 
![screenshot from 2018-10-26 10-15-41](https://user-images.githubusercontent.com/20714542/47535659-38295a00-d908-11e8-95fd-3f2dfbf1600f.png)
* we have to use `.. code-block:: C` which is fine here, but it also use `:class: no-execute`, I think we don't want to run it as well, so I leave it unchanged 
```
For example, consider the following C code, which sums the integers from 1 to 10

.. code-block:: c
    :class: no-execute

    #include <stdio.h>

    int main(void) {
        int i;
        int sum = 0;
        for (i = 1; i <= 10; i++) {
            sum = sum + i;
        }
        printf("sum = %d\n", sum);
        return 0;
    }

The variables ``i`` and ``sum`` are explicitly declared to be integers
```
- [x] ./rst_files/python_by_example.rst: :class: no-execute 
![screenshot from 2018-10-26 09-59-26](https://user-images.githubusercontent.com/20714542/47535134-f1d2fb80-d905-11e8-8ee2-696caf7dcc48.png)
```
This example helps to clarify how the ``for`` loop works:  When we execute a
loop of the form

.. code-block:: python3
    :class: no-execute

    for variable_name in sequence:
        <code block>

The Python interpreter performs the following:
```
- [x] ./rst_files/python_by_example.rst: :class: no-execute 
- [x] ./rst_files/python_by_example.rst: :class: no-execute 

With the list comprehension syntax, we can simplify the lines
![screenshot from 2018-10-26 10-01-55](https://user-images.githubusercontent.com/20714542/47535253-52623880-d906-11e8-8731-17b0a3f8f3cb.png)

```
.. code-block:: python3
    :class: no-execute

    ϵ_values = []
    for i in range(n):
        e = generator_type()
        ϵ_values.append(e)

into

.. code-block:: python3
    :class: no-execute

    ϵ_values = [generator_type() for i in range(n)]


```
- [x] ./rst_files/jv.rst: :class: no-execute 
![screenshot from 2018-10-26 09-55-58](https://user-images.githubusercontent.com/20714542/47535050-71ac9600-d905-11e8-8456-d2d39e196714.png)

```
 Set

.. code-block:: python3
    :class: no-execute

    K = 50
    plot_grid_max, plot_grid_size = 1.2, 100
    plot_grid = np.linspace(0, plot_grid_max, plot_grid_size)
    fig, ax = plt.subplots()
    ax.set_xlim(0, plot_grid_max)
    ax.set_ylim(0, plot_grid_max)


By examining the plot, argue that under the optimal policies, the state
```
- [x] ./rst_files/oop_intro.rst: :class: no-execute 
```
Consider the following expression 

.. code-block:: python3
    :class: no-execute

    '300' + 400
```

- [x] ./rst_files/oop_intro.rst:.. code-block:: none            #96

**Comment by natashawatkins** [comment]
@fkazemian could you please include these comments in separate PRs because it's a bit hard to follow the above comment. Thanks

**Comment by fkazemian** [comment]
sure

**Comment by fkazemian** [comment]
**python_advanced_features.rst**
for this lecture, If I change something I explained it on its PR #99, and I leave some tags unchanged, so I explained them here: 

- [x] ./rst_files/python_advanced_features.rst: :class: no-execute
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute
![screenshot from 2018-10-26 12-14-09](https://user-images.githubusercontent.com/20714542/47538782-d02f3f80-d918-11e8-9dca-6f4e289e4a6f.png)

the code is:
```
To see this, let's have another look at the :ref:`US cities data <us_cities_data>` 

.. code-block:: python3
    :class: no-execute

    f = open('us_cities.txt')
    f.__next__()
    
.. code-block:: none
    :class: no-execute
    
    'new york: 8244910\n'
    
.. code-block:: python3
    :class: no-execute

    f.__next__()
    
.. code-block:: none
    
    'los angeles: 3819702\n'

We see that file objects do indeed have a ``__next__`` method, and that calling this method returns the next line in the file

The next method can also be accessed via the builtin function ``next()``,
which directly calls this method

.. code-block:: python3
    :class: no-execute

    next(f)
```
If we run blocks, we get an error which we can't find the file to open it. so I think we should leave it in a no-execute way and show the results as a none block with no execute tags! Am I right?

---------------------------------------------------
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute   #99
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute   #99
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute
![screenshot from 2018-10-26 12-38-20](https://user-images.githubusercontent.com/20714542/47539417-1a65f000-d91c-11e8-838d-9564899b5724.png)
```
In fact this is how the ``for`` loop works:  If we write

.. code-block:: python3
    :class: no-execute

    for x in iterator:
        <code block>

```

- [x] ./rst_files/python_advanced_features.rst: :class: no-execute
![screenshot from 2018-10-26 12-39-50](https://user-images.githubusercontent.com/20714542/47539476-59944100-d91c-11e8-8415-654184a3b9be.png)
```
So now you know how this magical looking syntax works

.. code-block:: python3
    :class: no-execute

    f = open('somefile.txt', 'r')
    for line in f:
        # do something

```

- [x] ./rst_files/python_advanced_features.rst: :class: no-execute    #99
- [x] ./rst_files/python_advanced_features.rst: :class: no-execute    #99
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute
- [ ] ./rst_files/python_advanced_features.rst: :class: no-execute

**Comment by fkazemian** [comment]
- [x]  ./rst_files/stationary_densities.rst: :class: no-execute

```
.. code-block:: python3
    :class: no-execute

    initial_conditions = np.linspace(8, 0, J)



For each :math:`X_0` in this set,
```
![screenshot from 2018-10-26 11-51-49](https://user-images.githubusercontent.com/20714542/47538175-c9eb9400-d915-11e8-8196-c18c84f85a8d.png)

- [x] ./rst_files/stationary_densities.rst: :class: no-execute 
![screenshot from 2018-10-26 11-51-29](https://user-images.githubusercontent.com/20714542/47538145-9a3c8c00-d915-11e8-9c50-9bcd4f154fa3.png)
```
For the four initial distributions, use the shifted beta distributions

.. code-block:: python3
    :class: no-execute

    ψ_0 = beta(5, 5, scale=0.5, loc=i*2)

```

**Comment by fkazemian** [comment]
- [x] ./rst_files/python_oop.rst: :class: no-execute 
- [ ] ./rst_files/python_oop.rst: :class: no-execute 
![screenshot from 2018-10-26 11-47-51](https://user-images.githubusercontent.com/20714542/47538068-14204580-d915-11e8-8b6d-6b3d7b32d4e6.png)
it's fine, because by executing it we get a NameError  due to name 'ECDF' is not defined
------------------------------------
- [x] ./rst_files/python_essentials.rst: :class: no-execute  #98
- [x] ./rst_files/python_essentials.rst: :class: no-execute
![screenshot from 2018-10-26 11-16-09](https://user-images.githubusercontent.com/20714542/47537295-ab36ce80-d910-11e8-9ae9-9c19248d324f.png)
```
In this case you can shift the file to the pwd or specify the `full path <https://en.wikipedia.org/wiki/Path_%28computing%29>`_ to the file

.. code-block:: python3
    :class: no-execute

    f = open('insert_full_path_to_file/newfile.txt', 'r')
```

**Comment by fkazemian** [comment]
- [x] ./rst_files/python_oop.rst: :class: no-execute 
- [ ] ./rst_files/python_oop.rst: :class: no-execute 
![screenshot from 2018-10-26 11-47-51](https://user-images.githubusercontent.com/20714542/47538068-14204580-d915-11e8-8b6d-6b3d7b32d4e6.png)
it's fine, because by executing it we get a NameError  due to name 'ECDF' is not defined
------------------------------------
- [x] ./rst_files/python_essentials.rst: :class: no-execute  #98
- [x] ./rst_files/python_essentials.rst: :class: no-execute
![screenshot from 2018-10-26 11-16-09](https://user-images.githubusercontent.com/20714542/47537295-ab36ce80-d910-11e8-9ae9-9c19248d324f.png)
```
In this case you can shift the file to the pwd or specify the `full path <https://en.wikipedia.org/wiki/Path_%28computing%29>`_ to the file

.. code-block:: python3
    :class: no-execute

    f = open('insert_full_path_to_file/newfile.txt', 'r')
```

**Comment by fkazemian** [comment]
- [x] ./rst_files/sci_libs.rst: :class: no-execute 
![screenshot from 2018-10-26 10-18-31](https://user-images.githubusercontent.com/20714542/47535746-9e15e180-d908-11e8-94ec-4c524eabc396.png)
* we have to use `.. code-block:: C` which is fine here, but it also use `:class: no-execute`, I think we don't want to run it as well, so I leave it unchanged:
```
Here's a C function that will do the same thing

.. code-block:: c
    :class: no-execute

    double geo_prog(double alpha, int n) {
        double current = 1.0;
        double sum = current;
        int i;
        for (i = 1; i <= n; i++) {
            current = current * alpha;
            sum = sum + current;
        }
        return sum;
    }


If you're not familiar with C, the main thing you should take notice of is the
type definitions
```

**Comment by fkazemian** [comment]
- [x] ./rst_files/numba.rst: :class: no-execute 
![screenshot from 2018-10-26 10-15-41](https://user-images.githubusercontent.com/20714542/47535659-38295a00-d908-11e8-95fd-3f2dfbf1600f.png)
* we have to use `.. code-block:: C` which is fine here, but it also use `:class: no-execute`, I think we don't want to run it as well, so I leave it unchanged 
```
For example, consider the following C code, which sums the integers from 1 to 10

.. code-block:: c
    :class: no-execute

    #include <stdio.h>

    int main(void) {
        int i;
        int sum = 0;
        for (i = 1; i <= 10; i++) {
            sum = sum + i;
        }
        printf("sum = %d\n", sum);
        return 0;
    }

The variables ``i`` and ``sum`` are explicitly declared to be integers
```

**Comment by fkazemian** [comment]
- [x] ./rst_files/python_by_example.rst: :class: no-execute 
![screenshot from 2018-10-26 09-59-26](https://user-images.githubusercontent.com/20714542/47535134-f1d2fb80-d905-11e8-8ee2-696caf7dcc48.png)
```
This example helps to clarify how the ``for`` loop works:  When we execute a
loop of the form

.. code-block:: python3
    :class: no-execute

    for variable_name in sequence:
        <code block>

The Python interpreter performs the following:
```
- [x] ./rst_files/python_by_example.rst: :class: no-execute 
- [x] ./rst_files/python_by_example.rst: :class: no-execute 

With the list comprehension syntax, we can simplify the lines
![screenshot from 2018-10-26 10-01-55](https://user-images.githubusercontent.com/20714542/47535253-52623880-d906-11e8-8731-17b0a3f8f3cb.png)

```
.. code-block:: python3
    :class: no-execute

    ϵ_values = []
    for i in range(n):
        e = generator_type()
        ϵ_values.append(e)

into

.. code-block:: python3
    :class: no-execute

    ϵ_values = [generator_type() for i in range(n)]


```

**Comment by fkazemian** [comment]
- [x] ./rst_files/jv.rst: :class: no-execute 
![screenshot from 2018-10-26 09-55-58](https://user-images.githubusercontent.com/20714542/47535050-71ac9600-d905-11e8-8456-d2d39e196714.png)

```
 Set

.. code-block:: python3
    :class: no-execute

    K = 50
    plot_grid_max, plot_grid_size = 1.2, 100
    plot_grid = np.linspace(0, plot_grid_max, plot_grid_size)
    fig, ax = plt.subplots()
    ax.set_xlim(0, plot_grid_max)
    ax.set_ylim(0, plot_grid_max)


By examining the plot, argue that under the optimal policies, the state
```

**Comment by fkazemian** [comment]
Hi @natashawatkins , would you please review these PRs and comments? I leave something unchanged which I explained them in comments here(If there were no PR for them) and I also explain my changes as well. 
python_advanced_features                #99
schelling                                              #103
python_essentials                               #98
pandas.rst                                           #102
debugging                                           #97
finite_markov                                      #101
python_by_example                           #100
oop_intro                                            #96
stationary_densities           https://github.com/QuantEcon/lecture-source-py/issues/85#issuecomment-433279960
python_oop       https://github.com/QuantEcon/lecture-source-py/issues/85#issuecomment-433280122
sci_libs              https://github.com/QuantEcon/lecture-source-py/issues/85#issuecomment-433280645
numba              https://github.com/QuantEcon/lecture-source-py/issues/85#issuecomment-433280687
jv                       https://github.com/QuantEcon/lecture-source-py/issues/85#issuecomment-433280799

**Comment by natashawatkins** [comment]
@fkazemian I think you've misunderstood the tags - you want to use `:skip-test:` when the error should be generated and displayed, and `:no-execute:` when we don't want the error displayed. Could you be go through and fix the PRs accordingly?

**Comment by natashawatkins** [comment]
@fkazemian could you also please remove all the tags you've added to `.. code-block:: none` cells  - in some instances these cells should be completely removed if they are automatically generated

**Comment by mmcky** [comment]
this is the last remaining `high priority` issue to fix I think :)

**Comment by fkazemian** [comment]
Hi @natashawatkins @mmcky sure

**Comment by mmcky** [comment]
Closing this. Any unaddressed bugs will come up in the next comparison review.
